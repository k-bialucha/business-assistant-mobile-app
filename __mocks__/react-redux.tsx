import React from 'react';
import { Text, View } from 'react-native';

export const Provider: React.FC<{}> = ({ children }) => (
  <View>
    <Text>Provider</Text>
    <View>{children}</View>
  </View>
);

export const useDispatch = jest.fn().mockImplementation(() => jest.fn());

export const useSelector = jest.fn().mockImplementation(() => ({}));
