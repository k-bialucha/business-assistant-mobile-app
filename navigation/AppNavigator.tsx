import React from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';

import DashboardScreen from '../screens/DashboardScreen';
import SalesScreen from '../screens/SalesScreen';
import CostsScreen from '../screens/CostsScreen';
import CostsEntryScreen from '../screens/CostsEntryScreen';
import SalesEntryScreen from '../screens/SalesEntryScreen';

const DashboardStackNavigator = createStackNavigator();

const DashboardNavigator = () => {
  return (
    <DashboardStackNavigator.Navigator>
      <DashboardStackNavigator.Screen
        name="Dashboard"
        component={DashboardScreen}
      />
    </DashboardStackNavigator.Navigator>
  );
};

const SalesStackNavigator = createStackNavigator();

const SalesNavigator = () => {
  return (
    <SalesStackNavigator.Navigator>
      <SalesStackNavigator.Screen name="SalesList" component={SalesScreen} />
      <SalesStackNavigator.Screen
        name="SalesEntry"
        component={SalesEntryScreen}
      />
    </SalesStackNavigator.Navigator>
  );
};

const CostsStackNavigator = createStackNavigator();

const CostsNavigator = () => {
  return (
    <CostsStackNavigator.Navigator>
      <CostsStackNavigator.Screen name="CostsList" component={CostsScreen} />
      <CostsStackNavigator.Screen
        name="CostsEntry"
        component={CostsEntryScreen}
      />
    </CostsStackNavigator.Navigator>
  );
};

const AppTabNavigator = createBottomTabNavigator();

const AppNavigator = () => {
  return (
    <AppTabNavigator.Navigator>
      <AppTabNavigator.Screen name="Dashboard" component={DashboardNavigator} />
      <AppTabNavigator.Screen name="Sales" component={SalesNavigator} />
      <AppTabNavigator.Screen name="Costs" component={CostsNavigator} />
    </AppTabNavigator.Navigator>
  );
};

export default AppNavigator;
