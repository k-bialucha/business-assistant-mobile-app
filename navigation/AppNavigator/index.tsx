import React from 'react';
import { Platform } from 'react-native';

import { Ionicons } from '@expo/vector-icons';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { RouteProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';

import Colors from '../../theme/Colors';

import CostsNavigator from './CostsNavigator';
import DashboardNavigator from './DashboardNavigator';
import SalesNavigator from './SalesNavigator';
import SettingsNavigator from './SettingsNavigator';

export type RootStackParamList = {
  Dashboard: undefined;
  Sales: undefined;
  Costs: undefined;
  Settings: undefined;
};

export type ScreenProps<RouteName extends keyof RootStackParamList> = {
  navigation: StackNavigationProp<RootStackParamList, RouteName>;
  route: RouteProp<RootStackParamList, RouteName>;
};

const AppTabNavigator = createBottomTabNavigator<RootStackParamList>();

const AppNavigator: React.FC<{}> = () => {
  return (
    <AppTabNavigator.Navigator
      initialRouteName="Dashboard"
      screenOptions={({ route }) => ({
        tabBarIcon: ({ color, size }) => {
          const icons = {
            Dashboard: Platform.OS === 'android' ? 'md-stats' : 'ios-stats',
            Sales:
              Platform.OS === 'android' ? 'md-trending-up' : 'ios-trending-up',
            Costs:
              Platform.OS === 'android'
                ? 'md-trending-down'
                : 'ios-trending-down',
            Settings:
              Platform.OS === 'android' ? 'md-settings' : 'ios-settings',
          };

          return (
            <Ionicons name={icons[route.name]} size={size} color={color} />
          );
        },
      })}
      tabBarOptions={{
        showLabel: false,
        activeTintColor: Colors.navyBlue,
        inactiveTintColor: Colors.silver,
      }}
    >
      <AppTabNavigator.Screen name="Sales" component={SalesNavigator} />
      <AppTabNavigator.Screen name="Dashboard" component={DashboardNavigator} />
      <AppTabNavigator.Screen name="Costs" component={CostsNavigator} />
      <AppTabNavigator.Screen name="Settings" component={SettingsNavigator} />
    </AppTabNavigator.Navigator>
  );
};

export default AppNavigator;
