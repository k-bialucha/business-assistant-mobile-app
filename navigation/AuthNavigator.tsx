import React from 'react';

import { RouteProp } from '@react-navigation/native';
import {
  createStackNavigator,
  StackNavigationProp,
} from '@react-navigation/stack';

import LoginScreen from '../screens/LoginScreen';

export type ParamList = {
  Login: undefined;
};

const StackNavigator = createStackNavigator<ParamList>();

const AuthNavigator = () => {
  return (
    <StackNavigator.Navigator>
      <StackNavigator.Screen name="Login" component={LoginScreen} />
    </StackNavigator.Navigator>
  );
};

export default AuthNavigator;

// export type so screens can get typing
export type NavigationData<RouteName extends keyof ParamList> = {
  navigation: StackNavigationProp<ParamList, RouteName>;
  route: RouteProp<ParamList, RouteName>;
};
