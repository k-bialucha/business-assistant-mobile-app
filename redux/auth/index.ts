export { login, logout } from './actions';
export { LOGIN, LOGOUT, RequestStatus as LoginRequestStatus } from './types';

export { default } from './reducer';

export { default as authSaga } from './saga';
