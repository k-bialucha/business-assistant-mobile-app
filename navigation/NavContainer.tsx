import React from 'react';

import { NavigationContainer } from '@react-navigation/native';
import { useSelector } from 'react-redux';

import Loader from '../components/UI/Loader';
import { RequestStatus } from '../redux/auth/types';
import { RootState } from '../redux/rootReducer';
import StartupScreen from '../screens/StartupScreen';

import AppNavigator from './AppNavigator';
import AuthNavigator from './AuthNavigator';

const NavContainer = () => {
  const didTryAutoLogin: boolean = useSelector(
    (state: RootState) => !!state.auth.didTryAutoLogin
  );

  const isAuthenticated: boolean = useSelector(
    (state: RootState) => !!state.auth.token
  );

  const isLoading: boolean = useSelector(
    (state: RootState) => state.auth.requestStatus === RequestStatus.LOADING
  );

  return (
    <>
      <NavigationContainer>
        {isAuthenticated && <AppNavigator />}
        {!isAuthenticated && didTryAutoLogin && <AuthNavigator />}
        {!isAuthenticated && !didTryAutoLogin && <StartupScreen />}
      </NavigationContainer>
      <Loader isLoading={isLoading} />
    </>
  );
};

export default NavContainer;
