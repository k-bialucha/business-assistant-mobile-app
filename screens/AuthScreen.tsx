import React, { useState } from 'react';
import { Button } from 'react-native';

import { StackNavigationOptions } from '@react-navigation/stack';
import { useDispatch, useSelector } from 'react-redux';

import { NavigationData } from '../navigation/AuthNavigator';
import { login, LoginRequestStatus, logout } from '../redux/auth';
import { RootState } from '../redux/rootReducer';

import {
  StyledButtonsContainer,
  StyledInput,
  StyledInputContainer,
  StyledText,
  StyledView,
} from './AuthScreen.styled';

type Props = NavigationData<'Authorization'>;

const AuthScreen: React.FC<Props> = () => {
  const [username, setUsername] = useState<string>('');
  const [password, setPassword] = useState<string>('');

  const token: string = useSelector((state: RootState) => state.auth.token);
  const requestedUsername: string = useSelector(
    (state: RootState) => state.auth.username
  );
  const requestStatus: LoginRequestStatus = useSelector(
    (state: RootState) => state.auth.requestStatus
  );

  const dispatch = useDispatch();

  return (
    <StyledView>
      <StyledText success={token}>
        {'State: '}
        {requestStatus === LoginRequestStatus.UNAUTHENTICATED &&
          'unauthenticated'}
        {requestStatus === LoginRequestStatus.LOADING && '...'}
        {requestStatus === LoginRequestStatus.SUCCESS && 'SUCCESS!'}
        {requestStatus === LoginRequestStatus.FAILURE && 'FAILURE!'}
      </StyledText>
      <StyledText>Username: {requestedUsername || '?'}</StyledText>
      <StyledInputContainer>
        <StyledInput
          testID="username-input"
          value={username}
          placeholder="username"
          onChangeText={setUsername}
        />
      </StyledInputContainer>
      <StyledInputContainer>
        <StyledInput
          testID="password-input"
          value={password}
          placeholder="password"
          onChangeText={setPassword}
          secureTextEntry
        />
      </StyledInputContainer>
      <StyledButtonsContainer>
        <Button
          testID="login-button"
          title="Login"
          disabled={!!token}
          onPress={() => {
            dispatch(login(username, password));
          }}
        />
        <Button
          testID="logout-button"
          title="Logout"
          disabled={!token}
          onPress={() => {
            setUsername('');
            setPassword('');
            dispatch(logout());
          }}
        />
      </StyledButtonsContainer>
    </StyledView>
  );
};

export const AuthScreenNavOptions: StackNavigationOptions = {};

export default AuthScreen;
