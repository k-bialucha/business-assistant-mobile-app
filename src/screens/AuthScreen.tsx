import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';

import { FontAwesome } from '@expo/vector-icons';
import { StackNavigationOptions } from '@react-navigation/stack';
import { Button } from 'react-native-elements';
import { useDispatch } from 'react-redux';

import PlainText from '../components/UI/PlainText';
import { NavigationData } from '../navigation/AuthNavigator';
import { loginWithFacebook, loginWithGoogle } from '../redux/auth/actions';
import Colors from '../theme/Colors';

import {
  StyledButtonsContainer,
  StyledContainer,
  StyledLargeText,
  StyledLoginTextContainer,
  StyledWideContainer,
} from './AuthScreen.styled';

type Props = NavigationData<'Authorization'>;

const AuthScreen: React.FC<Props> = ({ navigation }) => {
  const dispatch = useDispatch();

  return (
    <StyledContainer>
      <View>
        <StyledLargeText>Make your business easier.</StyledLargeText>
      </View>
      <StyledButtonsContainer>
        <StyledWideContainer>
          <Button
            title="Create an account"
            buttonStyle={{ backgroundColor: Colors.gray }}
            onPress={() => navigation.navigate('Signup')}
          />
        </StyledWideContainer>
        <View
          style={{
            marginVertical: 15,
            backgroundColor: '#ffffff',
            height: 1,
            width: 50,
          }}
        />
        <StyledWideContainer>
          <Button
            title="Continue with Facebook"
            type="outline"
            buttonStyle={{ borderColor: Colors.silver }}
            titleStyle={{ color: Colors.silver, marginLeft: 10 }}
            icon={<FontAwesome name="facebook-f" size={18} color="white" />}
            onPress={() => {
              dispatch(loginWithFacebook());
            }}
          />
        </StyledWideContainer>
        <StyledWideContainer>
          <Button
            title="Continue with Google"
            type="outline"
            buttonStyle={{ borderColor: Colors.silver }}
            titleStyle={{ color: Colors.silver, marginLeft: 10 }}
            icon={<FontAwesome name="google" size={18} color="white" />}
            onPress={() => {
              dispatch(loginWithGoogle());
            }}
          />
        </StyledWideContainer>
      </StyledButtonsContainer>
      <StyledLoginTextContainer>
        <PlainText color="#ffffff">
          Already have an account?{' '}
          <TouchableWithoutFeedback
            onPress={() => navigation.navigate('Login')}
          >
            <PlainText theme="light" bold>
              Log in
            </PlainText>
          </TouchableWithoutFeedback>
        </PlainText>
      </StyledLoginTextContainer>
    </StyledContainer>
  );
};

export const AuthScreenNavOptions: StackNavigationOptions = {};

export default AuthScreen;
