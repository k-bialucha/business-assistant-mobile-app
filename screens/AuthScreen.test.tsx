import React from 'react';

import { shallow } from 'enzyme';
import { useSelector } from 'react-redux';

import { NavigationData } from '../navigation/AuthNavigator';
import { LoginRequestStatus } from '../redux/auth';

import AuthScreen from './AuthScreen';

type Props = NavigationData<'Authorization'>;

const fakeProps: Props = {
  route: { key: '1234', name: 'Authorization' },
  // @ts-ignore
  navigation: {},
};

jest.mock('react-redux', () => {
  return {
    __esModule: true,
    Provider: jest
      .fn()
      .mockImplementation(props => <div>{props.children}</div>),
    useSelector: jest
      .fn()
      .mockReturnValue({ token: 'some-token', username: 'some-user' }),
    useDispatch: jest.fn().mockImplementation(() => jest.fn()),
  };
});

describe('<AuthScreen />', () => {
  it('matches the snapshot', () => {
    (useSelector as jest.Mock).mockReturnValueOnce('some-token');
    (useSelector as jest.Mock).mockReturnValueOnce('some-user');
    (useSelector as jest.Mock).mockReturnValueOnce(LoginRequestStatus.SUCCESS);

    const tree = shallow(<AuthScreen {...fakeProps} />);

    expect(tree).toMatchSnapshot();
  });
});
