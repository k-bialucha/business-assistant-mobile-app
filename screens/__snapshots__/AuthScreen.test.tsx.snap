// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthScreen /> matches the snaphot 1`] = `
<Styled(View)>
  <View>
    <Styled(Text)>
      Make your business easier.
    </Styled(Text)>
  </View>
  <Styled(View)>
    <Styled(View)>
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "#5E6D8C",
          }
        }
        onPress={[Function]}
        title="Create an account"
      />
    </Styled(View)>
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": 1,
          "marginVertical": 15,
          "width": 50,
        }
      }
    />
    <Styled(View)>
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "borderColor": "#D5DCF2",
          }
        }
        icon={
          <Icon
            allowFontScaling={false}
            color="white"
            name="facebook-f"
            size={18}
          />
        }
        onPress={[Function]}
        title="Continue with Facebook"
        titleStyle={
          Object {
            "color": "#D5DCF2",
            "marginLeft": 10,
          }
        }
        type="outline"
      />
    </Styled(View)>
    <Styled(View)>
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "borderColor": "#D5DCF2",
          }
        }
        icon={
          <Icon
            allowFontScaling={false}
            color="white"
            name="google"
            size={18}
          />
        }
        onPress={[Function]}
        title="Continue with Google"
        titleStyle={
          Object {
            "color": "#D5DCF2",
            "marginLeft": 10,
          }
        }
        type="outline"
      />
    </Styled(View)>
  </Styled(View)>
  <Styled(View)>
    <PlainText
      color="#ffffff"
    >
      Already have an account?
       
      <TouchableWithoutFeedback
        onPress={[Function]}
      >
        <PlainText
          bold={true}
          light={true}
        >
          Log in
        </PlainText>
      </TouchableWithoutFeedback>
    </PlainText>
  </Styled(View)>
</Styled(View)>
`;
